File: src/main/kotlin/org/tendiwa/plane/geometry/straightSkeleton/ShrinkedNode.java
package org.tendiwa.plane.geometry.straightSkeleton;

import org.tendiwa.geometry.points.Point;

final class ShrinkedNode extends Node {
	ShrinkedNode(
			Point point,
			OriginalEdgeStart previousEdgeStart,
			OriginalEdgeStart currentEdgeStart
	) {
		super(point, previousEdgeStart, currentEdgeStart);
	}

	@Override
	boolean hasPair() {
		return false;
	}
}

File: src/main/kotlin/org/tendiwa/plane/geometry/straightSkeleton/ShrinkedNode.kt
package org.tendiwa.plane.geometry.straightSkeleton

import org.tendiwa.geometry.points.Point

internal class ShrinkedNode(
    point: Point,
    previousEdgeStart: OriginalEdgeStart,
    currentEdgeStart: OriginalEdgeStart) : Node(point, previousEdgeStart, currentEdgeStart) {

    internal override fun hasPair(): Boolean {
        return false
    }
}
